addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/LauncherApplication.java

```diff
@@ -31,63 +35,93 @@
  *
  * @author Sebastian Prehn - initial contribution
  */
+@NonNullByDefault
 public class LauncherApplication extends BaseChannelHandler<Launcher.AppInfoListener, LaunchSession> {
     private final Logger logger = LoggerFactory.getLogger(LauncherApplication.class);
+    private final Map<String, @Nullable List<AppInfo>> applicationListCache = new HashMap<>();
 
     private Launcher getControl(final ConnectableDevice device) {
         return device.getCapability(Launcher.class);
     }
 
     @Override
-    public void onReceiveCommand(@Nullable ConnectableDevice device, String channelId, LGWebOSHandler handler,
-            Command command) {
-        if (device == null) {
-            return;
-        }
-        if (device.hasCapabilities(Launcher.Application_List, Launcher.Application)) {
-            final String value = command.toString();
+    public void onDeviceReady(@NonNull ConnectableDevice device, @NonNull String channelId,
+            @NonNull LGWebOSHandler handler) {
+        super.onDeviceReady(device, channelId, handler);
+        if (hasCapability(device, logger, Launcher.Application_List)) {
+
             final Launcher control = getControl(device);
             control.getAppList(new Launcher.AppListListener() {
 
                 @Override
-                public void onError(ServiceCommandError error) {
-                    logger.warn("error requesting application list: {}.", error.getMessage());
+                public void onError(@Nullable ServiceCommandError error) {
+                    logger.warn("Error requesting application list: {}.", error == null ? "" : error.getMessage());
                 }
 
                 @Override
+                @NonNullByDefault({})
```

Comment:
Afaik we only use this for injected fields, why would you use it like this here?

Reply:
I otherwise get an error in Eclipse on line 62 no matter what method signature I use:  
public void onSuccess(List<AppInfo> appInfos) // wants @Nullable 
public void onSuccess(@Nullable List<AppInfo> appInfos) // complains that it cannot redefine the method